File: backend/app/routers/accommodations.py
from fastapi import APIRouter

from app.schemas.accommodation import AccommodationSearchRequest, AccommodationSearchResponse
from app.services import hotel_service

router = APIRouter(prefix="/accommodations", tags=["accommodations"])


@router.post("/search", response_model=AccommodationSearchResponse)
async def search(payload: AccommodationSearchRequest) -> AccommodationSearchResponse:
    results = await hotel_service.search_accommodations(payload.location, payload.checkin, payload.checkout)
    return AccommodationSearchResponse(results=results)

File: frontend/src/api/accommodations.ts
import { apiFetch } from './client'
import type { AccommodationSearchResponse } from '../types/api'

export function searchAccommodations(location: string, checkin: string, checkout: string) {
  return apiFetch<AccommodationSearchResponse>('/accommodations/search', {
    method: 'POST',
    body: JSON.stringify({ location, checkin, checkout }),
    auth: false,
  })
}
